src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/delete_sub_menu_test.cc

```diff
@@ -216,107 +391,112 @@ TEST_F(DeleteSubMenuRequestTest, Run_FindSubMenuFalse_UNSUCCESS) {
 }
 
 TEST_F(DeleteSubMenuRequestTest, Run_SendHMIRequest_SUCCESS) {
-  (*message_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
-  (*message_)[am::strings::params][am::strings::connection_key] =
-      kConnectionKey;
-
-  smart_objects::SmartObject sub_menu =
-      (*message_)[am::strings::msg_params][am::strings::menu_id];
-  EXPECT_CALL(app_mngr_, application(kConnectionKey)).WillOnce(Return(app_));
-  EXPECT_CALL(*app_, FindSubMenu(kMenuId)).WillOnce(Return(sub_menu));
+  auto sub_menu_ptr = GenerateSubMenuMap(kMenuId, 0);
+  RunDeleteSubMenuHappyPath(hmi_apis::FunctionID::UI_DeleteSubMenu);
+}
 
-  EXPECT_CALL(*app_, app_id()).WillOnce(Return(kConnectionKey));
-  EXPECT_CALL(app_mngr_, GetNextHMICorrelationID())
-      .WillOnce(Return(kCorrelationId));
+TEST_F(DeleteSubMenuRequestTest, Run_SendHMIRequestWithChilds_SUCCESS) {
+  auto sub_menu_ptr = GenerateSubMenuMap(kMenuId, 0);
+  auto sub_menu_child_ptr = GenerateSubMenuMap(kMenuIdChild, kMenuId);
+  auto vr_command_ptr = GenerateVRCommandMapChild(kCmdIdVr, kMenuIdChild);
+  auto ui_command_ptr = GenerateUICommandMapChild(kCmdIdUi, kMenuIdChild);
```

Comment:
Seems these variables are unused, either add these values to the `sub_menu_map_` or `commands_map_` outside of the functions, or change the functions to not return a value.

Reply:
@jacobkeeler values adding were moved outside the functions. The trick here is that `sub_menu_map_` and `commands_map` are holding just a raw pointer to smart objects, so we have to initialize a shared pointer and keep them alive till the end of the unit test. Otherwise, the allocated smart object gets destroyed after going out of scope and the map will contain a pointer to an invalid memory address.
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3748/commits/0b07d162a9f7cff42b61f6d6935d210d07246b7f